### PR TITLE
[src] Properly annotate designated initializers in many frameworks

### PR DESCRIPTION
**Pull request by spouliot**
Correspond to xtro `!missing-designated-initializer!` errors

### CODE REVIEW

**Comment by dalexsoto** [comment]
@spouliot can you post a link to a src API diff? I have just one doubt about generated ctors

**Comment by spouliot** [comment]
@dalexsoto sure can do
in the mean time bots produced the API diff -> https://jenkins.mono-project.com/job/xamarin-macios-pr-builder/5860/API_diff

**Comment by spouliot** [comment]
@dalexsoto https://gist.github.com/spouliot/d306e16aaf48ab82fab4559f9dbeb267
note that it's reversed, i.e. what I have locally being the _reference_

File: tests/xtro-sharpie/common-ModelIO.ignore [hunk 1]
```diff
@@ -5,18 +5,22 @@
...
+## we only export the overload that expose an NSError
 !missing-selector! MDLAsset::exportAssetToURL: not bound
+
+## those are just doing the same as the setComponent:forProtocol: and componentConformingToProtocol:
+## objectForKeyedSubscript:'s description is: "Allows shorthand [key] syntax for componentConformingToProtocol:"
 !missing-selector! MDLObject::objectForKeyedSubscript: not bound
 !missing-selector! MDLObject::setObject:forKeyedSubscript: not bound
+
+## no generator support for FastEnumeration - https://bugzilla.xamarin.com/show_bug.cgi?id=4391
+!missing-protocol-conformance! MDLAsset should conform to NSFastEnumeration
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/common-ModelIO.ignore]: Maybe we can hardcode NSFastEnumeration in xtro to skip it until we implement it? It seems a bit annoying to have to add exceptions when we know it won't work (and it's easy to detect).

Comment (VincentDondain) [reply tests/xtro-sharpie/common-ModelIO.ignore]: I'm not against special casing this and having xtro not report it. @spouliot do you want me to do that?

**Pull request merged**